tests/model_serving/model_server/maas_billing/utils.py

```diff
@@ -104,3 +109,67 @@ def llmis_name(client, namespace: str = "llm", label_selector: str | None = None
         raise RuntimeError("No Ready LLMInferenceService found")
 
     return service.name
+
+
+def patch_llmisvc_with_maas_router(
+    llm_service: LLMInferenceService,
+    client: DynamicClient,
+) -> None:
+    """
+    Patch an existing LLMInferenceService with MaaS router wiring and annotations.
+
+    This is used for TinyLlama so that the model is reachable via the maas-default-gateway
+
+    and participates in MaaS flows.
+    """
+    router_spec = {
+        "gateway": {
+            "refs": [
+                {
+                    "name": "maas-default-gateway",
+                    "namespace": "openshift-ingress",
+                }
+            ]
+        },
+        "route": {},
+    }
+
+    LOGGER.info(
+        f"MaaS LLMD: patching LLMInferenceService "
+        f"{llm_service.namespace}/{llm_service.name} "
+        f"with MaaS router spec: {router_spec}"
+    )
+
+    patch_body = {
+        "metadata": {
+            "annotations": {
+                "alpha.maas.opendatahub.io/tiers": "[]",
+            }
+        },
+        "spec": {
+            "router": router_spec,
+        },
+    }
+
+    llmisvc_res = client.resources.get(
+        api_version="serving.kserve.io/v1alpha1",
+        kind="LLMInferenceService",
+    )
+
+    try:
+        llmisvc_res.patch(
```

Comment:
Wouldn't it be better to use `ResourceEditor`? Example here: https://github.com/opendatahub-io/opendatahub-tests/blob/ba71c1c2803cb8ef90f9f478019e4b0920343964/tests/model_serving/model_server/kserve/components/conftest.py#L43

Or, alternatively, check if you can use `llm_service.api.patch` to simplify the code a little bit (you no longer would need `llmisvc_res`)

Reply:
+1 use ResourceEditor